doc/source/cd/background.ipynb

```diff
@@ -0,0 +1,608 @@
+    "\n",
+    "*Drift* is said to occur when the process underlying $\\mathbf{X}$ and $\\mathbf{Y}$ during deployment differs from the process that generated the training data. In this case, we can no longer expect the model's performance during deployment to match that observed on held out training data. If a *ground truth* is available during deployment, *supervised drift detection* can be performed, with the model's predictive performance monitored directly. However, in many deployment scenarios, such as the binary classification example below, a ground truth is not available and *unsupervised drift detection* methods are required.\n",
+    "\n",
+    "![drift in deployment](images/drift_deployment.png) \n",
+    "\n",
+    "To explore the different types of drift, consider the common scenario where we have input data $\\mathbf{X}$ and output data $\\mathbf{Y}$, jointly distributed according to $P(\\mathbf{X},\\mathbf{Y})$, and a model $f: x \\mapsto y$ trained on training/reference data with distribution $P_{ref}(\\mathbf{X},\\mathbf{Y})$. *Drift* is said to have occured when $P(\\mathbf{X},\\mathbf{Y}) \\ne P_{ref}(\\mathbf{X},\\mathbf{Y})$. Writing the joint distribution as\n",
+    "\n",
+    "$$\n",
+    "P(\\mathbf{X},\\mathbf{Y}) = P(\\mathbf{Y}|\\mathbf{X})P(\\mathbf{X}) = P(\\mathbf{X}|\\mathbf{Y})P(\\mathbf{Y}),\n",
+    "$$\n",
+    "\n",
+    "we can categorise drift into a number of types:\n",
+    "\n",
+    "- **Covariate drift**: Also referred to as input drift, this occurs when $P(\\mathbf{Y}|\\mathbf{X})$ = $P_{ref}(\\mathbf{Y}|\\mathbf{X})$, but the distribution of the input data has shifted $P(\\mathbf{X}) \\ne P_{ref}(\\mathbf{X})$. In a mortgage risk classification task, an example of this would be if at inference, applications for properties with more bedrooms than the properties present in the training data were considered. The model may now be extrapolating, potentially resulting in unreliable predictions. \n",
+    "\n",
+    "- **Prior drift**: Also referred to as label drift, this occurs when $P(\\mathbf{X}|\\mathbf{Y})=P_{ref}(\\mathbf{X}|\\mathbf{Y})$, but the distribution of the outputs has shifted $P(\\mathbf{Y}) \\ne P_{ref}(\\mathbf{Y})$. Using the same example again, an increase in the proportion of applications classsed as \"approved\" would be an example of prior drift. The model could still be accurate, but accuracy metrics might be affected. \n",
+    "\n",
+    "- **Concept drift**: This occurs when the process generating $y$ from $x$ has changed, such that $P(\\mathbf{Y}|\\mathbf{X}) \\ne P_{ref}(\\mathbf{Y}|\\mathbf{X})$. An example of this could be if a change in the wider economic climate resulted in a change in the risk for a given mortgage application. It is possible that the model might no longer give a suitable approximation of the true process. \n",
+    "\n",
```

Comment:
In this section we might already want to mention that prior/concept drift require access to ground truth which is typically not available so we need to find proxies (eg prediction distribution shift for prior drift).

Reply:
In general I quite like the symptoms/disease example from the introduction of [Lipton et al.](https://arxiv.org/abs/1802.03916) to illustrate the *P(X,Y)* breakdown.